src/components/video-texture-target.js

```diff
@@ -1,18 +1,86 @@
+ * it's textureNeedsUpdate property to true. Currently video-texture-target does this
+ * whenever its material is used during the primary camera render (as in, its in view).
+ */
+AFRAME.registerComponent("video-texture-source", {
+  schema: {
+    resolution: { type: "vec2", default: [1280, 720] },
+    fps: { default: 15 }
+  },
+
+  init() {
+    this.camera = findNode(this.el.object3D, n => n.isCamera);
+    this.camera.aspect = this.data.resolution[0] / this.data.resolution[1];
+
+    // TODO currently if a video-texture-source tries to render itself it will fail with a warning.
+    // If we want to support this we will need 2 render targets to swap back and forth.
+    this.renderTarget = new THREE.WebGLRenderTarget(this.data.resolution[0], this.data.resolution[1], {
+      format: THREE.RGBAFormat,
+      minFilter: THREE.LinearFilter,
+      magFilter: THREE.NearestFilter,
+      encoding: THREE.GammaEncoding,
+      depth: false,
+      stencil: false
+    });
+
+    const texture = this.renderTarget.texture;
+    texture.matrixAutoUpdate = false;
+    texture.matrix.scale(1, -1);
+    texture.matrix.translate(0, 1);
+
+    this.textureNeedsUpdate = false;
+  },
+
+  tock(time) {
+    if (!this.textureNeedsUpdate) return;
+
+    if (time < this.lastRenderTime + 1000 / this.data.fps) return;
+
+    // TODO handle hiding UI and showing first person head
+    // Once thats done also rework camrea-tool to use this to do its actual rendering
+
+    const sceneEl = this.el.sceneEl;
+    const renderer = this.renderer || sceneEl.renderer;
+
+    const tmpVRFlag = renderer.vr.enabled;
+    const tmpOnAfterRender = sceneEl.object3D.onAfterRender;
+    delete sceneEl.object3D.onAfterRender;
+    renderer.vr.enabled = false;
+
+    renderer.setRenderTarget(this.renderTarget);
+    renderer.render(sceneEl.object3D, this.camera);
+    renderer.setRenderTarget(null);
+
+    renderer.vr.enabled = tmpVRFlag;
+    sceneEl.object3D.onAfterRender = tmpOnAfterRender;
+
+    this.lastRenderTime = time;
+    this.textureNeedsUpdate = false;
+  }
+});
 
 /**
  * @component video-texture-target
  * This component is intended to be used on entities with a mesh/skinned mesh Object3D
  * The component swaps the base color map on the mesh's material with a video texture
- * Currently the video texture can only be a Webcam stream, but this component could be easily modified
- * to work with normal urls, screensharing, etc.
+ * Currently the video texture can come from a webcam stream or a camera entity with
+ * a video-texture-src component on it.
  */
 AFRAME.registerComponent("video-texture-target", {
   schema: {
     src: { type: "string" },
     targetBaseColorMap: { type: "boolean", default: true },
-    targetEmissiveMap: { type: "boolean", default: false }
+    targetEmissiveMap: { type: "boolean", default: false },
+    // TODO having both src and target is a bit odd
+    target: { type: "selector" }
```

Comment:
agree, `target` is a very confusing name, I think I just ended up blindly using that since I was looking at the trigger volume stuff as an example of being able to reference one node from another node in a gltf. I knew I was going to rework it anyway so didn't bother thinking of a better name.

My issue with having both `src` and `srcEl` it's pretty easy to create ambiguous states. Like if you set both `src` and `srcEl` which one should be preferred? Maybe `src` should be set to "srcEl" in this case or something... Right now the only actual usage of `src` is for video avatars which is completely internal (as in not in gltfs) and not networked, so it might also be fine to just rework that.